src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
Yes it makes sense. I have no idea what's the implications on runtime of any of these ideas though – would have to implement it and try. The beauty of new method is how trivial it is (only a few lines).

One pitfall if you try to be "too smart" is also that the splits end up being too nonrandom. You still need to have a fair amount of randomness to decorrelate the trees. There might be some smart way to achieve the same thing but I have no idea what it would be